### ISSUE TITLE
Direct payment to trampoline node fails

### ISSUE THREAD

**Issue by bitromortac** [spesmilo/electrum#7101]
In trampoline mode, when paying directly to the channel partner, the payment fails with a temporary node failure.

To reproduce:
On testnet, open a channel with endurance.
Do a payment to the starblocks shop (https://starblocks.acinq.co/).

The node backend of the starblocks shop is one of the default trampoline nodes. Presumably we send a trampoline onion (have not investigated) to the node and it doesn't know how to route to itself. We could also interpret this as an eclair bug, but if the receiving node is a trampoline node (our channel partner) we should just send a normal onion.

**Comment by SomberNight** [comment]
I can reproduce.
Could you have a look @ecdsa ?

<details>
 <summary>log</summary>

```
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0898
I | trampoline | created route with trampoline: fee_level=0, is legacy: True
I | trampoline | first trampoline: 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | trampoline | second trampoline: None
I | trampoline | params: {'fee_base_msat': 0, 'fee_proportional_millionths': 0, 'cltv_expiry_delta': 576}
I | trampoline | payload 0 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'outgoing_node_id': {'outgoing_node_id': b"\x03\x938\x84\xaa\xf1\xd6\xb1\x089~^\xfe\\\x86\xbc\xf2\xd8\xca\x8d/p\x0e\xda\x99\xdb\x92\x14\xfc'\x12\xb14"}, 'invoice_features': {'invoice_features': <LnFeatures.BASIC_MPP_OPT|PAYMENT_SECRET_OPT|VAR_ONION_OPT: 164352>}, 'invoice_routing_info': {'invoice_routing_info': b''}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
I | trampoline | payload 1 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
D | util.profiler | LNWallet.create_routes_for_payment 0.0078
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | lnpeer.pay len(route)=1
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] |   0: edge=1937413x40x0 hop_data=<OnionHopsDataSingle. is_tlv_payload=True. payload={'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941783}, 'payment_data': {'payment_secret': b"j\xd1/5\x96\xb5\xeb\n\xda\xd0\xe3\xc7\x14\xc9t+\x92h\xbbO.\x89\x88A\x85\x0c\x84'\xa6\x80\xd2\xed", 'total_msat': 150000, 'amount_msat': 150000}}. hmac=None>
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | adding trampoline onion to final payload
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. len(route)=1.
I | lnchannel.Channel | add_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. htlc: UpdateAddHtlc(amount_msat=150000, payment_hash=f1ebe24be28683dff3ddede148c07b37aa5ab0e1451962bc76a06384a2ca91d9, cltv_expiry=1941783, timestamp=1616380641, htlc_id=21)
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending UPDATE_ADD_HTLC
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 43.
I | lnchannel.Channel | sign_next_commitment 43
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0752
I | lnworker.LNWallet.[test_segwit_2] | amount inflight 150000
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 42
I | lnchannel.Channel | receive_revocation
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0751
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 43.
I | lnchannel.Channel | receive_new_commitment. ctn=43, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 42
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0722
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_update_fail_htlc. chan 1937413x40x0. htlc_id 21
I | lnchannel.Channel | receive_fail_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 44.
I | lnchannel.Channel | receive_new_commitment. ctn=44, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 43
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0728
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 44.
I | lnchannel.Channel | sign_next_commitment 44
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0627
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 43
I | lnchannel.Channel | receive_revocation
I | lnworker.LNWallet.[test_segwit_2] | htlc_failed (<OnionFailureCode.TRAMPOLINE_FEE_INSUFFICIENT: 8243>, b'')
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0667
I | lnworker.LNWallet.[test_segwit_2] | UPDATE_FAIL_HTLC. code=<OnionFailureCode.TRAMPOLINE_FEE_INSUFFICIENT: 8243>. decoded_data=None. data=
I | lnworker.LNWallet.[test_segwit_2] | error reported by 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | trampoline | created route with trampoline: fee_level=1, is legacy: True
I | trampoline | first trampoline: 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | trampoline | second trampoline: None
I | trampoline | params: {'fee_base_msat': 1000, 'fee_proportional_millionths': 100, 'cltv_expiry_delta': 576}
I | trampoline | payload 0 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'outgoing_node_id': {'outgoing_node_id': b"\x03\x938\x84\xaa\xf1\xd6\xb1\x089~^\xfe\\\x86\xbc\xf2\xd8\xca\x8d/p\x0e\xda\x99\xdb\x92\x14\xfc'\x12\xb14"}, 'invoice_features': {'invoice_features': <LnFeatures.BASIC_MPP_OPT|PAYMENT_SECRET_OPT|VAR_ONION_OPT: 164352>}, 'invoice_routing_info': {'invoice_routing_info': b''}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
I | trampoline | payload 1 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
D | util.profiler | LNWallet.create_routes_for_payment 0.0107
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | lnpeer.pay len(route)=1
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] |   0: edge=1937413x40x0 hop_data=<OnionHopsDataSingle. is_tlv_payload=True. payload={'amt_to_forward': {'amt_to_forward': 151015}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941783}, 'payment_data': {'payment_secret': b".J2g@\x80\x1fP\xb4\xbb\x87\x16\xee\xf9\xa45\x9a\x162\x1e\xcc\x93\x8a\xe0\x16\xf1\x9b'\xb0j\xd9V", 'total_msat': 151015, 'amount_msat': 151015}}. hmac=None>
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | adding trampoline onion to final payload
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. len(route)=1.
I | lnchannel.Channel | add_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. htlc: UpdateAddHtlc(amount_msat=151015, payment_hash=f1ebe24be28683dff3ddede148c07b37aa5ab0e1451962bc76a06384a2ca91d9, cltv_expiry=1941783, timestamp=1616380642, htlc_id=22)
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending UPDATE_ADD_HTLC
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 45.
I | lnchannel.Channel | sign_next_commitment 45
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0603
I | lnworker.LNWallet.[test_segwit_2] | amount inflight 150000
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 44
I | lnchannel.Channel | receive_revocation
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0714
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 45.
I | lnchannel.Channel | receive_new_commitment. ctn=45, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 44
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0651
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_update_fail_htlc. chan 1937413x40x0. htlc_id 22
I | lnchannel.Channel | receive_fail_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 46.
I | lnchannel.Channel | receive_new_commitment. ctn=46, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 45
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0679
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 46.
I | lnchannel.Channel | sign_next_commitment 46
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0652
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 45
I | lnchannel.Channel | receive_revocation
I | lnworker.LNWallet.[test_segwit_2] | htlc_failed (<OnionFailureCode.TEMPORARY_NODE_FAILURE: 8194>, b'')
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0755
I | lnworker.LNWallet.[test_segwit_2] | UPDATE_FAIL_HTLC. code=<OnionFailureCode.TEMPORARY_NODE_FAILURE: 8194>. decoded_data={}. data=
I | lnworker.LNWallet.[test_segwit_2] | error reported by 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | trampoline | created route with trampoline: fee_level=1, is legacy: True
I | trampoline | first trampoline: 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | trampoline | second trampoline: None
I | trampoline | params: {'fee_base_msat': 1000, 'fee_proportional_millionths': 100, 'cltv_expiry_delta': 576}
I | trampoline | payload 0 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'outgoing_node_id': {'outgoing_node_id': b"\x03\x938\x84\xaa\xf1\xd6\xb1\x089~^\xfe\\\x86\xbc\xf2\xd8\xca\x8d/p\x0e\xda\x99\xdb\x92\x14\xfc'\x12\xb14"}, 'invoice_features': {'invoice_features': <LnFeatures.BASIC_MPP_OPT|PAYMENT_SECRET_OPT|VAR_ONION_OPT: 164352>}, 'invoice_routing_info': {'invoice_routing_info': b''}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
I | trampoline | payload 1 {'amt_to_forward': {'amt_to_forward': 150000}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941207}, 'payment_data': {'payment_secret': b"\xfa\xf9\xb8\xb3\xee\x98\x01r\xa4^\x10D\xf7\xad*\xf2s1\xda4}\x0ek\xea\xe3,\xcd'\x82\xfdm6", 'total_msat': 150000}}
D | util.profiler | LNWallet.create_routes_for_payment 0.0117
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | lnpeer.pay len(route)=1
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] |   0: edge=1937413x40x0 hop_data=<OnionHopsDataSingle. is_tlv_payload=True. payload={'amt_to_forward': {'amt_to_forward': 151015}, 'outgoing_cltv_value': {'outgoing_cltv_value': 1941783}, 'payment_data': {'payment_secret': b"\xcf#\xf0\x87\x86\x14Gl\xf3\x8e\xd3r\xa0.\xd3\x83\xe1\x97Sj9'\xe68\xb3\xdf\x00\xc9\xdb0]\xbd", 'total_msat': 151015, 'amount_msat': 151015}}. hmac=None>
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | adding trampoline onion to final payload
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. len(route)=1.
I | lnchannel.Channel | add_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | starting payment. htlc: UpdateAddHtlc(amount_msat=151015, payment_hash=f1ebe24be28683dff3ddede148c07b37aa5ab0e1451962bc76a06384a2ca91d9, cltv_expiry=1941783, timestamp=1616380643, htlc_id=23)
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending UPDATE_ADD_HTLC
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 47.
I | lnchannel.Channel | sign_next_commitment 47
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0815
I | lnworker.LNWallet.[test_segwit_2] | amount inflight 150000
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 46
I | lnchannel.Channel | receive_revocation
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0645
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 47.
I | lnchannel.Channel | receive_new_commitment. ctn=47, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 46
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0627
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_update_fail_htlc. chan 1937413x40x0. htlc_id 23
I | lnchannel.Channel | receive_fail_htlc
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_commitment_signed. chan 1937413x40x0. ctn: 48.
I | lnchannel.Channel | receive_new_commitment. ctn=48, len(htlc_sigs)=0
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_revoke_and_ack. chan 1937413x40x0. ctn: 47
I | lnchannel.Channel | revoke_current_commitment
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0841
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending REVOKE_AND_ACK
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | send_commitment. chan 1937413x40x0. ctn: 48.
I | lnchannel.Channel | sign_next_commitment 48
I | transaction | adding signature for 0247e41b3dc5524089d60c37902baaaec3328919e66ab64100853351a9d8aef6b3
D | transaction | is_complete False
D/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | Sending COMMITMENT_SIGNED
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0983
I/P | lnpeer.Peer.[LNWallet, 76.223.71.211:9735] | on_revoke_and_ack. chan 1937413x40x0. ctn: 47
I | lnchannel.Channel | receive_revocation
I | lnworker.LNWallet.[test_segwit_2] | htlc_failed (<OnionFailureCode.TEMPORARY_NODE_FAILURE: 8194>, b'')
I | storage.WalletStorage | saved c:\users\user\appdata\roaming\electrum\testnet\wallets\test_segwit_2
D | util.profiler | WalletDB._write 0.0672
I | lnworker.LNWallet.[test_segwit_2] | UPDATE_FAIL_HTLC. code=<OnionFailureCode.TEMPORARY_NODE_FAILURE: 8194>. decoded_data={}. data=
I | lnworker.LNWallet.[test_segwit_2] | error reported by 03933884aaf1d6b108397e5efe5c86bcf2d8ca8d2f700eda99db9214fc2712b134
I | lnworker.LNWallet.[test_segwit_2] | payment failure: PaymentFailure('TEMPORARY_NODE_FAILURE')
D | util.profiler | WalletDB._write 0.0000
```
</details>

**Comment by ecdsa** [comment]
can you try this? I don't have testnet coins right now

```
@@ -1435,21 +1435,29 @@ class LNWallet(LNWorker):
                 for chan in active_channels:
                     if not self.is_trampoline_peer(chan.node_id):
                         continue
-                    trampoline_onion, amount_with_fees, cltv_delta = create_trampoline_route_and_onion(
-                        amount_msat=amount_msat,
-                        total_msat=final_total_msat,
-                        min_cltv_expiry=min_cltv_expiry,
-                        my_pubkey=self.node_keypair.pubkey,
-                        invoice_pubkey=invoice_pubkey,
-                        invoice_features=invoice_features,
-                        node_id=chan.node_id,
-                        r_tags=r_tags,
-                        payment_hash=payment_hash,
-                        payment_secret=payment_secret,
-                        local_height=local_height,
-                        trampoline_fee_level=trampoline_fee_level,
-                        use_two_trampolines=use_two_trampolines)
-                    trampoline_payment_secret = os.urandom(32)
+                    if chan.node_id == invoice_pubkey:
+                        trampoline_onion = None
+                        trampoline_payment_secret = payment_secret
+                        trampoline_total_msat = final_total_msat
+                        amount_with_fees = amount_msat
+                        cltv_delta = min_cltv_expiry
+                    else:
+                        trampoline_onion, amount_with_fees, cltv_delta = create_trampoline_route_and_onion(
+                            amount_msat=amount_msat,
+                            total_msat=final_total_msat,
+                            min_cltv_expiry=min_cltv_expiry,
+                            my_pubkey=self.node_keypair.pubkey,
+                            invoice_pubkey=invoice_pubkey,
+                            invoice_features=invoice_features,
+                            node_id=chan.node_id,
+                            r_tags=r_tags,
+                            payment_hash=payment_hash,
+                            payment_secret=payment_secret,
+                            local_height=local_height,
+                            trampoline_fee_level=trampoline_fee_level,
+                            use_two_trampolines=use_two_trampolines)
+                        trampoline_payment_secret = os.urandom(32)
+                        trampoline_total_msat = amount_with_fees
                     if chan.available_to_spend(LOCAL, strict=True) < amount_with_fees:
                         continue
                     route = [
@@ -1462,7 +1470,7 @@ class LNWallet(LNWorker):
                             cltv_expiry_delta=0,
                             node_features=trampoline_features)
                     ]
-                    routes = [(route, amount_with_fees, amount_with_fees, amount_msat, cltv_delta, trampoline_payment_secret, trampoline_onion)]
+                    routes = [(route, amount_with_fees, trampoline_total_msat, amount_msat, cltv_delta, trampoline_payment_secret, trampoline_onion)]
                     break
```

**Comment by SomberNight** [comment]
The patch works.
I've tested paying both the direct peer (starblocks), and some other node (htlc.me) as well.

**Comment by ecdsa** [comment]
thanks. merged in 0f4297cfa82e2056504a60a174f00b93016d6432
